### PR TITLE
Adding test vectors for Secp256k1

### PR DESCRIPTION
**Pull request by nitro-neal**
# Overview
Adding test vectors for Secp256k1

# Description
Web5TestVectorsCryptoEs256k
sign()
verify()

Will get checkmarks for kt here:
https://tbd54566975.github.io/sdk-report-runner/

### CODE REVIEW

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt [hunk 1]
```diff
@@ -79,4 +86,91 @@ class Secp256k1Test {
...
+      }
+    }
+  }
+
+  @Test
+  fun verify() {
+    val typeRef = object : TypeReference<TestVectors<VerifyTestInput, Boolean>>() {}
+    val testVectors = mapper.readValue(File("../web5-spec/test-vectors/crypto_es256k/verify.json"), typeRef)
+
+    testVectors.vectors.filter { it.errors != true }.forEach { vector ->
```
Comment (jiyoontbd) [inline crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:138]:
```suggestion
    testVectors.vectors.filter { it.errors == false }.forEach { vector ->
```

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt [hunk 2]
```diff
@@ -79,4 +86,91 @@ class Secp256k1Test {
...
+  )
+
+  private val mapper = jacksonObjectMapper()
+
+  @Test
+  fun sign() {
+    val typeRef = object : TypeReference<TestVectors<SignTestInput, String>>() {}
+    val testVectors = mapper.readValue(File("../web5-spec/test-vectors/crypto_es256k/sign.json"), typeRef)
+
+    testVectors.vectors.filter { it.errors != true }.forEach { vector ->
```
Comment (jiyoontbd) [inline crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:110]:
```suggestion
    testVectors.vectors.filter { it.errors == false }.forEach { vector ->
```

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt [hunk 3]
```diff
@@ -79,4 +86,91 @@ class Secp256k1Test {
...
+      val inputByteArray: ByteArray = hexStringToByteArray(vector.input.data)
+      val jwkMap = vector.input.key
+      val signatureByteArray = hexStringToByteArray(vector.input.signature)
+
+      val ecJwk = ECKey.parse(jwkMap.toString())
+
+      if (vector.output == true) {
+        assertDoesNotThrow { Secp256k1.verify(ecJwk, inputByteArray, signatureByteArray)  }
+      } else {
+        assertFails { Secp256k1.verify(ecJwk, inputByteArray, signatureByteArray)  }
```
[the remark this answers is not in the document]
Comment (nitro-neal) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:148]: so there is failing like an exception is thrown, failing like the verify is false, and success

[the remark this answers is not in the document]
Comment (jiyoontbd) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:148]: hmmm... i would maybe say this lib then needs to be re-evaluated :) but yea, i think that makes sense for now.

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt [hunk 4]
```diff
@@ -79,4 +86,91 @@ class Secp256k1Test {
...
+  private fun hexStringToByteArray(s: String): ByteArray {
+    val len = s.length
+    val data = ByteArray(len / 2)
+    for (i in 0 until len step 2) {
+      data[i / 2] = ((Character.digit(s[i], 16) shl 4) + Character.digit(s[i + 1], 16)).toByte()
+    }
+    return data
+  }
+
+  private fun byteArrayToHexString(bytes: ByteArray): String {
```
Comment (phoebe-lew) [inline crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:173]: Would these be used anywhere else? i.e should they be extracted somewhere shareable?

Comment (phoebe-lew) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt:173]: also surprised it's not available via `common`!

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt [hunk 5]
```diff
@@ -79,4 +86,91 @@ class Secp256k1Test {
...
+        val jwkMap = vector.input.key
+        val signatureByteArray = hexStringToByteArray(vector.input.signature)
+
+        val ecJwk = ECKey.parse(jwkMap.toString())
+        Secp256k1.verify(ecJwk, inputByteArray, signatureByteArray)
+      }
+    }
+  }
+
+  private fun hexStringToByteArray(s: String): ByteArray {
```
[the remark this answers is not in the document]
Comment (nitro-neal) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt]: I changed to use the apache Hex lib

File: crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt (hunk 4 above)
Comment (nitro-neal) [reply crypto/src/test/kotlin/web5/sdk/crypto/Secp256k1Test.kt]: I changed to use the apache Hex lib

**Review by decentralgabe — APPROVED** [review]

**Pull request merged**